### DISCUSSION TITLE
May I ask how to solve my problem with the nan value?

### DISCUSSION THREAD

**Discussion by Yikxinxin** [yalmip/YALMIP#1274] [General discussion]
May I ask how to solve my problem with the nan value?
 Each parameter in the outer objective function is a fixed value, but the value of the function displays nan value.
![1683595388426](https://user-images.githubusercontent.com/104471890/236970776-a0091e2d-ae71-415b-907a-cee7ba20a150.png)
 Give the result as follows：
![1683595622294](https://user-images.githubusercontent.com/104471890/236970977-ec7de7c1-f580-4357-80a7-7de01c133600.png)
 I called one of the values in the formula and had a result, but directly outputting the value of the formula displayed nan value.
 The result of t=1：
![1683596142073](https://user-images.githubusercontent.com/104471890/236972056-290029c7-ee45-4cf1-a4ad-7fc9bfd531b3.png)
I don't know how to solve it, I hope to receive your help. Thank you very much!

**Comment by johanlofberg** [comment]
my guess R is initially a double and then you start trying to insert sdpvar expressions into it
https://yalmip.github.io/naninmodel

**Reply by Yikxinxin** [reply]
Thank you for your reply！
I omitted the previous step and directly defined the outer objective function OI, but the result is still NAN.
![1](https://github.com/yalmip/YALMIP/assets/104471890/5c605aaf-da9b-476b-a197-d3ed12222160)
And, I will use the method you guided to represent the inner function. First, create a zero matrix, and then assign F (t, i) to it, and the result is still NAN.
![2](https://github.com/yalmip/YALMIP/assets/104471890/949cd191-39f6-4c2d-af9d-6847978a9d2d)
May I ask how to express the objective function?

**Reply by johanlofberg** [reply]
It's explained in the link. Don't assign into doubles, create by concatenation instead.

(and the code now is just a complete failure, as you keep clearing F inside the loop so even if it was possible to insert sdpvars into doubles, the code would fail)

**Reply by Yikxinxin** [reply]
Thank you very much for your patient answer！
